tests/TestConfiguration.ci.php

```diff
@@ -42,4 +42,18 @@
 defined('TESTS_ZEND_CACHE_MEMCACHED_ENABLED') || define('TESTS_ZEND_CACHE_MEMCACHED_ENABLED', true);
 defined('TESTS_ZEND_CACHE_LIBMEMCACHED_ENABLED') || define('TESTS_ZEND_CACHE_LIBMEMCACHED_ENABLED', true);
 
+/**
+ * Zend_Db_Adapter_Pdo_Mysql and Zend_Db_Adapter_Mysqli
+ *
+ * There are separate properties to enable tests for the PDO_MYSQL adapter and
+ * the native Mysqli adapters, but the other properties are shared between the
+ * two MySQL-related Zend_Db adapters.
+ */
+defined('TESTS_ZEND_DB_ADAPTER_PDO_MYSQL_ENABLED') || define('TESTS_ZEND_DB_ADAPTER_PDO_MYSQL_ENABLED',  true);
```

Comment:
This problem remains:
- https://github.com/zf1s/zf1/pull/34#issuecomment-737886457

The problem happens on 5.4 and 5.5 builds only, and it's flaky, not appearing in the same test:

```
2021-01-27T13:17:38.2309404Z 1) Zend_Db_Select_MysqliTest::testSelectLimitFetchCol
2021-01-27T13:17:38.2311953Z Zend_Db_Statement_Mysqli::_execute(): Error occurred while closing statement
2021-01-27T13:17:38.2313926Z
2021-01-27T13:17:38.2316415Z /home/runner/work/php-zf1s/php-zf1s/packages/zend-db/library/Zend/Db/Statement/Mysqli.php:220
2021-01-27T13:17:38.2319081Z /home/runner/work/php-zf1s/php-zf1s/packages/zend-db/library/Zend/Db/Statement.php:303
2021-01-27T13:17:38.2321732Z /home/runner/work/php-zf1s/php-zf1s/packages/zend-db/library/Zend/Db/Adapter/Abstract.php:480
2021-01-27T13:17:38.2324356Z /home/runner/work/php-zf1s/php-zf1s/packages/zend-db/library/Zend/Db/Adapter/Abstract.php:794
2021-01-27T13:17:38.2326989Z /home/runner/work/php-zf1s/php-zf1s/tests/Zend/Db/Select/TestCommon.php:1470
2021-01-27T13:17:38.2328976Z


2021-01-27T13:17:51.3030428Z 1) Zend_Db_Select_MysqliTest::testSelectLimitFetchCol
2021-01-27T13:17:51.3032499Z Zend_Db_Statement_Mysqli::_execute(): Error occurred while closing statement
2021-01-27T13:17:51.3033998Z
2021-01-27T13:17:51.3036132Z /home/runner/work/php-zf1s/php-zf1s/packages/zend-db/library/Zend/Db/Statement/Mysqli.php:220
2021-01-27T13:17:51.3038380Z /home/runner/work/php-zf1s/php-zf1s/packages/zend-db/library/Zend/Db/Statement.php:303
2021-01-27T13:17:51.3040658Z /home/runner/work/php-zf1s/php-zf1s/packages/zend-db/library/Zend/Db/Adapter/Abstract.php:480
2021-01-27T13:17:51.3043752Z /home/runner/work/php-zf1s/php-zf1s/packages/zend-db/library/Zend/Db/Adapter/Abstract.php:794
2021-01-27T13:17:51.3046714Z /home/runner/work/php-zf1s/php-zf1s/tests/Zend/Db/Select/TestCommon.php:1470
2021-01-27T13:17:51.3048992Z


2021-01-28T07:20:07.5157100Z 1) Zend_Db_Select_MysqliTest::testSelectLimitNone
2021-01-28T07:20:07.5159164Z Zend_Db_Adapter_Abstract::_quoteIdentifierAs(): Error occurred while closing statement
2021-01-28T07:20:07.5160738Z
2021-01-28T07:20:07.5162771Z /home/runner/work/php-zf1s/php-zf1s/packages/zend-db/library/Zend/Db/Adapter/Abstract.php:1007
2021-01-28T07:20:07.5165063Z /home/runner/work/php-zf1s/php-zf1s/packages/zend-db/library/Zend/Db/Adapter/Abstract.php:961
2021-01-28T07:20:07.5167234Z /home/runner/work/php-zf1s/php-zf1s/tests/Zend/Db/TestUtil/Mysqli.php:78
2021-01-28T07:20:07.5169281Z /home/runner/work/php-zf1s/php-zf1s/tests/Zend/Db/TestUtil/Common.php:131
2021-01-28T07:20:07.5171341Z /home/runner/work/php-zf1s/php-zf1s/tests/Zend/Db/TestUtil/Common.php:83
2021-01-28T07:20:07.5173439Z /home/runner/work/php-zf1s/php-zf1s/tests/Zend/Db/TestUtil/Common.php:495
2021-01-28T07:20:07.5175502Z /home/runner/work/php-zf1s/php-zf1s/tests/Zend/Db/TestUtil/Mysqli.php:124
2021-01-28T07:20:07.5177528Z /home/runner/work/php-zf1s/php-zf1s/tests/Zend/Db/TestSetup.php:64
2021-01-28T07:20:07.5179009Z


2021-01-28T07:21:48.5751951Z 1) Zend_Db_Select_MysqliTest::testSelectLimitFetchCol
2021-01-28T07:21:48.5756739Z Zend_Db_Statement_Mysqli::_execute(): Error occurred while closing statement
2021-01-28T07:21:48.5760618Z
2021-01-28T07:21:48.5764142Z /home/runner/work/php-zf1s/php-zf1s/packages/zend-db/library/Zend/Db/Statement/Mysqli.php:220
2021-01-28T07:21:48.5768148Z /home/runner/work/php-zf1s/php-zf1s/packages/zend-db/library/Zend/Db/Statement.php:303
2021-01-28T07:21:48.5772154Z /home/runner/work/php-zf1s/php-zf1s/packages/zend-db/library/Zend/Db/Adapter/Abstract.php:480
2021-01-28T07:21:48.5776190Z /home/runner/work/php-zf1s/php-zf1s/packages/zend-db/library/Zend/Db/Adapter/Abstract.php:794
2021-01-28T07:21:48.5780097Z /home/runner/work/php-zf1s/php-zf1s/tests/Zend/Db/Select/TestCommon.php:1478
2021-01-28T07:21:48.5782754Z
```

- https://github.com/glensc/php-zf1s/runs/1776916345?check_suite_focus=true
- https://github.com/glensc/php-zf1s/runs/1782104058?check_suite_focus=true

Reply:
Previously I figured, I could use this skip marker:

```php
        if (getenv("CI") && $this instanceof \Zend_Db_Table_Select_MysqliTest && PHP_VERSION_ID >= 50400 && PHP_VERSION_ID < 50600) {
            $this->markTestSkipped('Unexpected ext/mysql error in CI: Error occurred while closing statement (PHP 5.4/5.5)');
        }
```

But as it happens with any `$this->_db->select()` (or the like), I can't just mark each of the tests as skipped?

@falkenhawk advice on how to proceed.

Reply:
Argh, this is not going to be easy... I could not find any reliable answer to why mysqli is crashing like that, and what's worse pointing to random code lines like in https://github.com/zf1s/zf1/pull/34#issuecomment-737886457

Could it be that `mysqli` is simply broken on php 5.4/5.5? Or perhaps mysql 5.7 is too new for it and not fully compatible? 🤔

Reply:
It's not related to MySQL server version, I tried different versions in https://github.com/zf1s/zf1/pull/34 (maybe the commits even visible from there)

And in that issue I already pointed out it's coming from php-src:
- https://github.com/zf1s/zf1/pull/34#issuecomment-738010890

So the PHP provided by setup-php behaves that, there's nothing we can do, unless you want to patch the PHP or use own build of PHP. The PHP source of setup-php is rather weird, only 5.3 is built from source, the other versions come from random .deb packages from various sources. I helped to setup the PHP 5.3 build to use docker there: https://github.com/shivammathur/php5-ubuntu/pull/1. i think that project should build all versions from source, but someone has to step in and do that, so far I've only proposed it: https://github.com/shivammathur/php5-ubuntu/pull/1#issuecomment-753980535

Reply:
Seems I already gave my thought on the same topic in the previous PR:
- https://github.com/zf1s/zf1/pull/34#issuecomment-740513160

Reply:
so what could we do now 🤔 skip zend/db/mysql tests on 5.4 and 5.5?

Reply:
Yes, I think so. turning off mysqli tests for those broken platforms. it's not like we lose much, as 5.3 and 5.6 should cover those platforms.

Reply:
using `@`-operator, like `@parent::setUp()` also helped. but it helped only partially, i.e if the error came from `setUp` method.

Reply:
Changed so that these versions are skipped on GHA and travis tests only those versions.

meanwhile, can work on improving setup-php base images:
- https://github.com/shivammathur/php5-ubuntu/pull/1#issuecomment-773695461

Reply:
Disabled ext/mysqli testing for 5.4/5.5, can enable again when better php build is available.

Reply:
@falkenhawk why 👍 only, not a merge? what's missing?

Reply:
I'll merge it later today, I'm afk sometimes and only checking quickly on mobile